versionedController/v1/configuration/configuration.go

```diff
@@ -53,79 +57,82 @@ func (configurationController *Controller) Put(c *gin.Context) {
+	configurationController.updateToK8s(c, configModel)
+}
 
-	if configurationModel.ClientID != nil && configurationModel.EnableGithub != nil {
-		credentialType := c.Query("type")
-		var data map[string]string
-
-		switch credentialType {
-		case string(models.GithubAuth):
-			{
-				data = map[string]string{
-					types.GITHUB_CLIENT_ID:     *configurationModel.ClientID,
-					types.GITHUB_CLIENT_SECRET: *configurationModel.ClientSecret,
-				}
-			}
-		default:
-			{
-				c.Writer.WriteHeader(http.StatusBadRequest)
-				return
-			}
-		}
-
-		secret, err := k8s.ClientSet.CoreV1().Secrets(types.DefaultNamespace).Get(c.Request.Context(), types.Credentials, metav1.GetOptions{})
-		if err != nil {
-			c.String(http.StatusInternalServerError, "Error storing credentials : %v", err)
-			log.Errorln("Error getting secret ", err)
-			return
-		}
-
-		secret.StringData = data
-		_, err = k8s.ClientSet.CoreV1().Secrets(types.DefaultNamespace).Update(c.Request.Context(), secret, metav1.UpdateOptions{})
-		if err != nil {
-			c.String(http.StatusInternalServerError, "Error storing credentials : %v", err)
-			log.Errorln("Error updating secret ", err)
-			return
-		}
-
-		controller.Server.GithubConfig.ClientID = *configurationModel.ClientID
-		controller.Server.GithubConfig.ClientSecret = *configurationModel.ClientSecret
+// updateToK8s updates the value of Oauth secrets in secrets & configmaps
+// TODO: Need to lock this operation, so only one request can update it at a time
+// TODO(vharsh): Use pflags to configure all of these
+func (configurationController *Controller) updateToK8s(c *gin.Context, requestModel *Model) {
+	// 1. Get current configuration from configmap
+	// 2. See if any update is required
+	// 3. Respond back with updated *Model
+	cm, err := k8s.ClientSet.CoreV1().ConfigMaps(types.DefaultNamespace).
+		Get(c.Request.Context(), types.DefaultConfigMap, metav1.GetOptions{})
+	if err != nil {
+		c.JSON(http.StatusPreconditionFailed, gin.H{
+			"message": "Unable to persist config change",
+		})
 	}
-	if configurationModel.EnableGithub != nil {
-		cm, err := k8s.ClientSet.CoreV1().ConfigMaps(types.DefaultNamespace).Get(c.Request.Context(), types.DefaultConfigMap, metav1.GetOptions{})
-		if err != nil {
-			c.String(http.StatusInternalServerError, "Error enabling github login : %v", err)
-			log.Errorln("Error getting configmap ", err)
-			return
-		}
-
-		cm.Data[types.DISABLE_GITHUBAUTH] = strconv.FormatBool(!*configurationModel.EnableGithub)
-		_, err = k8s.ClientSet.CoreV1().ConfigMaps(types.DefaultNamespace).Update(c.Request.Context(), cm, metav1.UpdateOptions{})
-		if err != nil {
-			c.String(http.StatusInternalServerError, "Error enabling github login : %v", err)
-			log.Errorln("Error updating configmap ", err)
-			return
-		}
-
-		controller.Server.Config.DisableGithubAuth = !*configurationModel.EnableGithub
+	githubEnable, _ := strconv.ParseBool(cm.Data[types.DISABLE_GITHUBAUTH])
+	githubEnable = !githubEnable
```

Comment:
Why are we doing this assignment?

Reply:
In the configmap we have the key, `DISABLE_GITHUBAUTH`, maybe I'll use different variable name(githubDisable) to better reflect the meaning, these did seem like a shoddy practice when I was writing it.